cibuildwheel/util.py

```diff
@@ -76,3 +76,32 @@ def download(url, dest):
             file.write(response.read())
     finally:
         response.close()
+
+
+class DependencyConstraints(object):
+    def __init__(self, base_file_path):
+        assert os.path.exists(base_file_path)
+        self.base_file_path = os.path.abspath(base_file_path)
+
+    @classmethod
+    def with_defaults(cls):
+        return cls(
+            base_file_path=os.path.join(os.path.dirname(__file__), 'resources', 'constraints.txt')
+        )
+
+    def get_for_python_version(self, version):
```

Comment:
Good idea 👍

Reply:
Actually, our tooling can't generate such a file, since packages don't contain pypy compability metadata. (an implementation of Python 2.7 is assumed to be compatible between CPython and Pypy). So our tooling must generate a Python 2 file that's used by both implementations of Python 2.7. 

At that point we're either

1. generating both a `constraints-cp27.txt` and `constraints-pp27.txt` with the same contents
2. generating a `constraints-python27.txt` but adding code to look for an implementation-specific file first.

I prefer (2), but if that's the design, we can add that later if we need it, and keep the design simpler for now.

Reply:
Hmmmmpf, yeah. Maybe it's too soon to worry about this.

But binary packages do contain metadata on whether they are meant for PyPy or CPython, no? Then again, these are not the ones we need in there, as `pip`, ... are all source distributions.

I'm a little bit worried after the `virtualenv` troubles of #274, that it might be good to be able to pin a PyPy version (even if that file almost always contains the same as the CPython file). But I'm not as convinced as when I wrote the first comment, as we haven't really bumped into a situation where we'd need it, yet.

Reply:
> But binary packages do contain metadata on whether they are meant for PyPy or CPython, no?

They do, but a source distribution is normally also available, which in theory would be a fallback for every platform.